libs/near-indexer/src/near-indexer.service.ts

```diff
@@ -479,4 +449,45 @@ export class NearIndexerService {
 
     return queryBuilder;
   }
+
+  async findAccountChangeActionsByContractName(
+    contractName: string,
+    fromBlockTimestamp?: number,
+  ): Promise<AccountChange[]> {
+    return this.buildAccountChangeActionQuery(
+      contractName,
+      fromBlockTimestamp,
+    ).getMany();
+  }
+
+  private buildAccountChangeActionQuery(
+    contractName: string,
+    fromBlockTimestamp?: number,
+  ): SelectQueryBuilder<AccountChange> {
+    return this.accountChangeRepository
+      .createQueryBuilder('account_change')
+      .leftJoinAndSelect('account_change.causedByReceipt', 'receipts')
+      .leftJoinAndSelect('receipts.originatedFromTransaction', 'transactions')
+      .leftJoinAndSelect(
+        'transactions.transactionAction',
+        'transaction_actions',
+      )
+      .leftJoinAndSelect('receipts.receiptActions', 'action_receipt_actions')
+      .leftJoin(
+        'execution_outcomes',
+        'execution_outcomes',
+        'execution_outcomes.receipt_id = receipts.receipt_id',
+      )
+      .where('account_change.affected_account_id like :id', {
+        // Need to find all DAOs + factory contract changes
+        id: `%${contractName}`,
+      })
+      .andWhere('execution_outcomes.status != :failStatus', {
```

Comment:
if account was changed does it mean that execution outcome was success?

Reply:
no in some cases there is Failure statuses for account changes receipts for some reason :(